onnxruntime/core/graph/ep_api_types.cc

```diff
@@ -110,12 +110,42 @@ Status EpNode::Create(const Node& node, const EpGraph* ep_graph,
   const auto& node_attrs = node.GetAttributes();
   std::unordered_map<std::string, std::unique_ptr<ONNX_NAMESPACE::AttributeProto>> ep_node_attributes_map;
   std::vector<OrtOpAttr*> ep_node_attributes;
+  std::unordered_map<std::string, std::unique_ptr<OrtValue>> tensor_attribute_values;
+
+  auto tensor_proto_to_ort_value = [&](const ONNX_NAMESPACE::TensorProto& tensor_proto,
+                                       std::string& tensor_proto_name) -> Status {
+    const auto& graph_viewer = ep_graph->GetGraphViewer();
+
+    // Initialize OrtValue for tensor attribute.
+    // Note: using std::unique_ptr<OrtValue> because we return a OrtValue* to the user and we want it to be stable.
+    auto tensor_attribute_value = std::make_unique<OrtValue>();
+    AllocatorPtr tensor_attribute_allocator = CPUAllocator::DefaultInstance();
+    ORT_RETURN_IF_ERROR(utils::TensorProtoToOrtValue(Env::Default(), graph_viewer.ModelPath(), tensor_proto,
+                                                     tensor_attribute_allocator, *tensor_attribute_value));
+
+    tensor_attribute_values.emplace(tensor_proto_name, std::move(tensor_attribute_value));
+
+    return Status::OK();
+  };
 
   if (node_attrs.size() > 0) {
     ep_node_attributes.reserve(node_attrs.size());
 
     for (const auto& item : node_attrs) {
       auto attr = std::make_unique<ONNX_NAMESPACE::AttributeProto>(item.second);  // Copy AttributeProto and owned by this EpNode object.
+
+      // Create and cache an OrtValue for the 'TENSOR' attribute
+      if (attr->type() == onnx::AttributeProto::TENSOR) {
+        const auto& tensor_proto = attr->t();
+        // Some tensor proto could have no name.
+        // Create a name for that case since we need the name as the key for lookup later.
+        std::string tensor_proto_name = tensor_proto.name();
+        if (tensor_proto.name().empty()) {
```

Comment:
if the tensor_proto_name is **not** empty, is it guaranteed to be unique in the model?